packages/fxa-auth-client/lib/utils.ts

```diff
@@ -39,3 +39,18 @@ export function base64UrlToUint8(value: string): Uint8Array {
 export function xor(array1: Uint8Array, array2: Uint8Array) {
   return new Uint8Array(array1.map((byte, i) => byte ^ array2[i]));
 }
+
+export function mergeHeaders(a?: Headers, b?: Headers) {
+  if (b == null) {
+    b = new Headers();
+  }
+
+  if (a == null) {
+    a = new Headers();
+  }
+
+  const result = new Headers();
+  a.forEach((v, k) => result.set(k, v));
+  b.forEach((v, k) => result.set(k, v));
+  return result;
+}
```

Comment:
Interesting! How would this be different from, say:

```js
export function mergeHeaders(a = new Headers(), b = new Headers()) {
  for (const [k, v] of b.entries()) {
    a.set(k, v);
  }
  return a;
}
```

If we default the arguments to a `new Headers()` instance, we can cut 6 lines. If we just append `b` into `a` (or `a` into `b` &mdash; depending on if you want the first or last value to "win" &mdash; versus creating a third `Headers()` instance; we can also avoid a `.forEach()` loop.

There was also a version of the [`Headers()` constructor](https://developer.mozilla.org/en-US/docs/Web/API/Headers#examples) that lets you pass an array of arrays (like a Map) to create a new instance, but I couldn't get that to work locally, so I don't think it liked me trying to destructure that `.entries()` iterator.

Reply:
Also, is something using this function [yet]? This is the only reference to `mergeHeaders()` that I can find.

Reply:
Ah, this was just a quick util function that I was using during debugging. Didn't mean to check it in...